entrypoint.sh

```diff
@@ -1,46 +1,52 @@
 #!/bin/sh
 
-touch ${SSH_KEY_FILE:=/id_rsa}
-chmod 0400 ${SSH_KEY_FILE:=/id_rsa}
+# Set up key file
+KEY_FILE=${SSH_KEY_FILE:=/id_rsa}
+if [ ! -f "${KEY_FILE}" ]; then
+	echo "[ERROR] No SSH Key file found"
+	exit 1
+fi
+eval $(ssh-agent -s)
+cat "${SSH_KEY_FILE}" | ssh-add -k -
 
+# Set up known_hosts file if needed
 STRICT_HOSTS_KEY_CHECKING=no
 KNOWN_HOSTS=${SSH_KNOWN_HOSTS:=/known_hosts}
 if [ -f "${KNOWN_HOSTS}" ]; then
-    chmod 0400 ${KNOWN_HOSTS}
-    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS}"
+    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS} -o CheckHostIP=no "
```

Comment:
I dislike this as default.  This is forcing everyone who uses to a KNOWN_HOST arg to ignore IP.  A very insecure practice by default.

I would just prefer a new argument at this point.  `AUTOSSH_ARGS`, which is by default blank tacked in the middle of the command somewhere.

Reply:
Which naturally should come with a documentation update. **cough cough**